### ISSUE TITLE
use `char` or `uint8` instead; cuchar is deprecated [Deprecated]

### ISSUE THREAD

**Issue by arkanoid87** [PMunch/futhark#13]
I see this warning `use `char` or `uint8` instead; cuchar is deprecated [Deprecated]` in my Futhark based projects, so I went down the rabbit hole.

https://github.com/PMunch/futhark/blob/875a7f5cf727483bc4c4d394f7d3c23699002aa8/src/opir.nim#L43

from IRC #llvm:
```
there are three kinds of char. `char`, `signed char` and `unsigned char`
plain old `char` may either be signed or unsigned. Regardless which it is, it is a distinct type from `signed char` and `unsigned char`
`UChar` is `unsigned char`
`Char_U` is a `char` which just so happens to be `unsigned`.
more specifically, `Char_U` is a `char` which can represent the same values as `unsigned char` but is a distinct type from it.
sidebar, this exists because some platforms preferred their char to be unsigned vs signed. the standard makes it possible for a programmer to specify the type when it is important to them.
The GCC flag for overriding this behavior is -funsigned-char/-fsigned-char. The MSVC flag is /J.
```

**Comment by PMunch** [comment]
I guess we could change it to `uint8`, but I used cuchar to make the wrapper as true to the original as possible.

**Comment by PMunch** [comment]
This is fixed as of version 0.9.0
